django/contrib/contenttypes/fields.py

```diff
@@ -160,20 +160,33 @@ def _check_content_type_field(self):
     def get_cache_name(self):
         return self.name
 
-    def get_content_type(self, obj=None, id=None, using=None):
+    def get_content_type(self, obj=None, id=None, using=None, model=None):
         if obj is not None:
             return ContentType.objects.db_manager(obj._state.db).get_for_model(
                 obj, for_concrete_model=self.for_concrete_model
             )
         elif id is not None:
             return ContentType.objects.db_manager(using).get_for_id(id)
+        elif model is not None:
+            return ContentType.objects.get_for_model(
+                model, for_concrete_model=self.for_concrete_model
+            )
         else:
             # This should never happen. I love comments like this, don't you?
             raise Exception("Impossible arguments to GFK.get_content_type!")
 
     def get_prefetch_queryset(self, instances, queryset=None):
-        if queryset is not None:
-            raise ValueError("Custom queryset can't be used for this lookup.")
+        if queryset is not None and not isinstance(queryset, dict):
+            raise ValueError("Custom queryset must be a dict for this lookup.")
```

Comment:
Do we need to use a dict? It seems unnecessary complicated. Model classes that we pass in the keys must match the base models from querysets. We also don't protect against incorrect values e.g.
```python
queryset={
    Animal: Bookmark.objects.all()
}
```
I would use a list/tuple instead and raise an error when a queryset for the specific model is already resolved, e.g.
```python
for qs in querysets:
    ct_id = self.get_content_type(model=qs.query.model, using=qs.db).pk
    if ct_id in custom_queryset_dict:
        raise ValueError(...)
    custom_queryset_dict[ct_id] = qs
```
We should also add a new argument (maybe `querysets`) because it's misleading to pass list of querysets in the argument called `queryset`.